sdks/java/io/iceberg/src/test/java/org/apache/beam/sdk/io/iceberg/IcebergIOReadTest.java

```diff
@@ -350,6 +350,40 @@ public void testScanSelectedFields() throws Exception {
     testPipeline.run();
   }
 
+  @Test
+  public void testScanWithFilter() throws Exception {
+    TableIdentifier tableId =
+        TableIdentifier.of("default", "table" + Long.toString(UUID.randomUUID().hashCode(), 16));
+    Table simpleTable = warehouse.createTable(tableId, TestFixtures.SCHEMA);
+    final Schema schema = icebergSchemaToBeamSchema(TestFixtures.SCHEMA);
+
+    List<List<Record>> expectedRecords = warehouse.commitData(simpleTable);
+
+    IcebergIO.ReadRows read =
+        IcebergIO.readRows(catalogConfig()).from(tableId).withFilter("\"id\" < 10");
```

Comment:
Let's also add unit tests for a filter that use multiple operations ?

Reply:
Done